src/index.ts

```diff
@@ -1,7 +1,94 @@
-import { SingleSpaAureliaOptions } from './types';
+import { LifeCycles } from 'single-spa';
 
-export default function singleSpaAurelia(options: SingleSpaAureliaOptions): void {
-  options.bootstrap(() => {
-    return 10;
-  });
+import { chooseDomElementGetter, getHost } from './dom';
+import { SingleSpaAureliaOptions, BootstrappedSingleSpaAureliaOptions } from './types';
+
+const defaultOptions: BootstrappedSingleSpaAureliaOptions = {
+  component: null,
+  aurelia: null!,
+  Aurelia: null!,
+  JitHtmlBrowserConfiguration: null!,
+  domElementGetter: undefined,
+  configure: undefined,
+};
+
+function bootstrap(): Promise<void> {
+  return Promise.resolve();
+}
+
+function mount(options: BootstrappedSingleSpaAureliaOptions, props: any): Promise<unknown> {
+  const domElementGetter = chooseDomElementGetter(options, props);
+
+  if (typeof domElementGetter !== 'function') {
+    throw new Error(`
+      single-spa-aurelia: Cannot mount Aurelia application '${
+        props.name || props.appName
+      }' without a domElementGetter provided either as an option or prop
+    `);
+  }
+
+  const host = getHost(domElementGetter);
+
+  const aurelia = new options.Aurelia(options.container).register(
+    options.JitHtmlBrowserConfiguration,
+  );
+
+  if (typeof options.configure === 'function') {
+    options.configure(aurelia, props);
+  }
+
+  options.aurelia = aurelia;
+
+  return aurelia
+    .app({
+      host,
+      component: options.component,
+    })
+    .start()
+    .wait()
+    .then(() => {
+      if (aurelia.root.controller?.bindingContext) {
```

Comment:
When would the controller or bindingContext be undefined?

Reply:
Oh that actually the compiler was whining that those properties are `IController | undefined` thus it encouraged me to put `if` condition.